metaflow/cli.py

```diff
@@ -101,6 +110,82 @@ def logger(body='', system_msg=False, head='', bad=False, timestamp=True):
                 fg=LOGGER_BAD_COLOR if bad else None)
 
 
+def get_ordered_steps(graph, type='linear'):
+    """
+    Returns the ordered step names in the graph (FlowGraph) from start step to end step as a list of strings containing the
+    step names.
+
+    Note: All MF graphs start at the "start" step and end with the "end" step (both of which are mandatory).
+    """
+
+    ordered_steps = ['start']
+    current_node_name = 'start'
+
+    # This is not an ideal way to iterate over the graph, but it's the (easy+)right thing to do for now.
+    # This may need to change as work on improvements.
+    while current_node_name != 'end':
+        for node in graph:
+            if node.name == current_node_name:
+                if current_node_name != 'end':
+                    current_node_name = node.out_funcs[0]
+                    ordered_steps.append(current_node_name)
+                    break
+    return ordered_steps
+
+
+def run_step_op(step_name, code_url=DEFAULT_FLOW_CODE_URL):
+    """
+    Method to create a kfp container op to run a single step (here, we also execute our custom pre-start step
+    for setup as we aren't maintaining state) and then execute the given step.
+    """
+
+    python_cmd = """ "python helloworld.py --datastore-root ", $1, " step {} --run-id ", $2, " --task-id ", $4, " --input-paths", $2"/"$5"/"$6 """.format(
+        step_name)
+    command_inside_awk = """ {{ print {0} }}""".format(python_cmd)
+    final_run_cmd = """ python helloworld.py pre-start | awk 'END{}' | sh """.format(command_inside_awk)
+
+    return dsl.ContainerOp(
+
+        name='StepRunner-{}'.format(step_name),
+        image='ai-docker.artifactory.zgtools.net/artificial-intelligence/ai-platform/aip-infrastructure/dockerfiles/python-debian/python-debian:2.1.6b34e4f3',
```

Comment:
let's use an image accessible by anyone on the net, like `python:alpine3.6`

Reply:
Changed to a public image on docker hub. Following our discussions, I had to create my own. So the I've uploaded and am using right now is a copy of the aip-kfp-example image